x/evm/keeper/grpc_query.go

```diff
@@ -256,6 +258,12 @@ func (k Keeper) EthCall(c context.Context, req *types.EthCallRequest) (*types.Ms
 
 // EstimateGas implements eth_estimateGas rpc api.
 func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*types.EstimateGasResponse, error) {
+	return k.EstimateGasInternal(c, req, types.RPC)
+}
+
+// EstimateGasInternal call from internal and rpc is diffrence. when call from rpc client, gas kv config is needs
+// to be zero to be same as executing the msg.
```

Comment:
```suggestion
```suggestion
// EstimateGasInternal returns the gas estimation for the corresponding request. 
// This function is called from the RPC client (eth_estimateGas) and internally 
// in CallEVMWithData function in the erc20 keeper.
// When calling from the RPC client, we need to reset the gas meter before
// simulating the transaction to have
// an accurate gas estimation for EVM extension transactions
```
```